fix: OPTIC-2017: Select follow ups

yyassi-heartex opened the pull request:
This pull request introduces several updates across the `CreateProject` page and the `Select` component, focusing on improving functionality, user experience, and code maintainability. The most significant changes include adding a new `selectFirstIfEmpty` prop to the `Select` component, updating styles and behavior for the `CreateProject` page, and renaming a prop for clarity.

### Updates to the `Select` component:
* Added a new `selectFirstIfEmpty` prop to automatically select the first option if no initial value is provided. This includes updates to the logic for determining the initial value and handling state changes (`web/libs/ui/src/lib/select/select.tsx`). [[1]](diffhunk://#diff-567459e612a12d17fb4237f0381f79f4148ad291939afbf005a1c02e398fe942R43) [[2]](diffhunk://#diff-567459e612a12d17fb4237f0381f79f4148ad291939afbf005a1c02e398fe942L52-R55) [[3]](diffhunk://#diff-567459e612a12d17fb4237f0381f79f4148ad291939afbf005a1c02e398fe942R73-R83)
* Renamed the `isInProgress` prop to `isLoading` for better clarity and consistency (`web/libs/ui/src/lib/select/types.ts`).
* Removed unused CSS classes related to group headings in the `Option` component to simplify styles (`web/libs/ui/src/lib/select/select.tsx`).

### Updates to the `CreateProject` page:
* Adjusted the `Caption` element to display as a block for better layout control (`web/apps/labelstudio/src/pages/CreateProject/CreateProject.scss`).
* Minor text adjustment in the `Caption` element to ensure proper spacing in the rendered output (`web/apps/labelstudio/src/pages/CreateProject/CreateProject.jsx`).

yyassi-heartex committed:
fix: OPTIC-2017: Select follow ups

yyassi-heartex committed:
slight tweak for project creation label

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-2017/followup

yyassi-heartex committed:
slight visual tweak to the way the text renders

bmartel commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -49,7 +50,9 @@ export const Select = forwardRef(
     const valueRef = useRef<any>();
     let initialValue = defaultValue?.value ?? defaultValue ?? externalValue?.value ?? externalValue;
-
+    if (selectFirstIfEmpty && !initialValue) {
```
`false` is a valid option here
```suggestion
    if (selectFirstIfEmpty && initialValue === undefined) {
```

yyassi-heartex replied:
good call!

bmartel commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -67,9 +70,17 @@ export const Select = forwardRef(
     }, [externalValue, multiple]);
 
+    useEffect(() => {
+      if (valueRef.current || !selectFirstIfEmpty || !options?.[0]) return;
```
```suggestion
      if (valueRef.current || !selectFirstIfEmpty || options?.[0] === null || options?.[0] === undefined) return;
```

yyassi-heartex committed:
Update web/libs/ui/src/lib/select/select.tsx

Co-authored-by: bmartel <brandonmartel@gmail.com>

bmartel commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -49,7 +50,9 @@ export const Select = forwardRef(
     const valueRef = useRef<any>();
     let initialValue = defaultValue?.value ?? defaultValue ?? externalValue?.value ?? externalValue;
-
+    if (selectFirstIfEmpty && initialValue !== null || initialValue !== undefined) {
```
```suggestion
    if (selectFirstIfEmpty && initialValue === null || initialValue === undefined) {
```
Sorry I messed up the logic originally here

yyassi-heartex committed:
Update web/libs/ui/src/lib/select/select.tsx

Co-authored-by: bmartel <brandonmartel@gmail.com>

yyassi-heartex committed:
clearing up ambiguity

yyassi-heartex committed:
Update web/libs/ui/src/lib/select/select.tsx

Co-authored-by: bmartel <brandonmartel@gmail.com>

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-2017/followup

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14636618782)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14636618782

borisheartex committed:
Merge branch 'develop' into 'fb-optic-2017/followup'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14636618782

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-2017/followup

nass600 commented:
@yyassi-heartex Shouldn't `selectFirstIfEmpty` functionality be handled by the caller? Do we have many cases where this happens so it deserves a prop?

yyassi-heartex commented:
> @yyassi-heartex Shouldn't `selectFirstIfEmpty` functionality be handled by the caller? Do we have many cases where this happens so it deserves a prop?

yeah this is needed in LSE ModelVersionSelector - there is a very specific use case where it relies on the first option being selected if no value is provided

bmartel commented:
@yyassi-heartex I was going to ask the same question as @nass600, because we really don't want to bake in a functionality that is necessitated by a bad upstream implementation, if anything we would handle that at the location site of the Select and not force Select to have domain specific implementation details.

yyassi-heartex committed:
resolving confusing focus states

yyassi-heartex committed:
improved defaultValue behavior

yyassi-heartex commented:
@nass600 @bmartel take another look - i moved away from the selectFirstOption prop - leveraging defaultValue instead - let me know your thoughts

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14647408690)
> Successfully merged:  2 files changed, 30 insertions(+), 4 deletions(-)

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-2017/followup'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14647408690

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14662056634)

borisheartex committed:
Merge branch 'develop' into 'fb-optic-2017/followup'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14662056634

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14662056634

yyassi-heartex committed:
making integration tests a little less flakey

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-2017/followup

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14702312118)

borisheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14702312118

borisheartex committed:
Merge branch 'develop' into 'fb-optic-2017/followup'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14702312118

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-2017/followup
